src/frontend/apps/impress/src/features/docs/doc-share/component/DocShareModal.tsx

```diff
@@ -147,102 +162,139 @@ export const DocShareModal = ({ doc, onClose }: Props) => {
     });
   };
 
+  const getListHeight = () => {
+    const contentHeight = 690;
+    const inputHeight = 70;
+    const marginTop = 11;
+
+    const footerHeight = footerRef.current?.clientHeight ?? 0;
+    const selectedUsersHeight = selectedUsersRef.current?.clientHeight ?? 0;
+
+    return (
+      contentHeight -
+      footerHeight -
+      selectedUsersHeight -
+      inputHeight -
+      marginTop
+    );
+  };
+
+  const handleRef = (node: HTMLDivElement | null) => {
+    if (!node) {
+      return;
+    }
+
+    footerRef.current = node;
+    setListHeight(getListHeight());
+  };
+
   return (
-    <Modal
-      isOpen
-      closeOnClickOutside
-      data-testid="doc-share-modal"
-      aria-label={t('Share modal')}
-      size={isDesktop ? ModalSize.LARGE : ModalSize.FULL}
-      onClose={onClose}
-      title={<Box $align="flex-start">{t('Share the document')}</Box>}
-    >
-      <Box
+    <>
+      <Modal
+        isOpen
+        closeOnClickOutside
+        data-testid="doc-share-modal"
         aria-label={t('Share modal')}
-        $direction="column"
-        $justify="space-between"
+        size={isDesktop ? ModalSize.LARGE : ModalSize.FULL}
+        onClose={onClose}
+        title={<Box $align="flex-start">{t('Share the document')}</Box>}
       >
         <Box
-          $flex={1}
-          className="toto"
-          $css={css`
-            overflow-y: auto;
-            [cmdk-list] {
-              overflow-y: auto;
-              height: ${isDesktop
-                ? '400px'
-                : 'calc(100vh - 49px -  68px - 229px)'};
-            }
-          `}
+          className="share-modal-container"
```

Comment:
You don't need a classname if you include <ShareModalStyle /> inside the Modal componant.